### PR TITLE
Organise Chat Page

### LINKED ISSUE #14: Chat page
**Describe the task that needs to be done.**
Need to create a chat list page for users to chat with each others

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
create a page component which consist of the chat list component and a search bar

**Notes**
Block by the creation of the chat list component and search bar component

### PR DESCRIPTION
**Pull request by colaMeowSY**
## Description
Organised chat page and made most components responsive resize for desktop. Linking to API and socket io is yet to be done.
The responsive resize layout for mobile is yet to be decided.
Fixes #14 

## How has this been tested?
Tests will be written after linking the page with API and socket io.

## Screenshots
*Provide screenshots of what has been implemented. Leave blank if not applicable*
Bigger window view:
ChatPage

Smaller window view:

Resized_ChatPage

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by DanilLinkov — APPROVED** [review]
Looks good to me nice job 👍

File: src/pages/Chat/Chat.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    courseLink: 'https://random.link',
+  },
+];
+
+function Chat() {
+  const [chatList,setChatList] = useState([]);
+
+  useEffect(()=>{
+       const getChatList = async () => {
+         try {const response = await getBuddyChatList();
```
Comment (uoa-lzha989) [inline src/pages/Chat/Chat.jsx:36]: Very finnicky point, but if you get a chance, I think this is meant to be reformated?

Comment (colaMeowSY) [reply src/pages/Chat/Chat.jsx:36]: I'll run it through prettier later :)

File: src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap [hunk 2]
```diff
@@ -24,11 +26,15 @@ exports[`DialoguesList produces correct snapshot of multiple messages 1`] = `
     <div
       className="bubble me bubble-img"
     >
-      <img
-        alt="imageMessage"
-        className="content-img"
-        src="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
-      />
+      <p
```
Comment (uoa-lzha989) [inline src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap:29]: Ideally would replace this with the Typography import from MUI :)

File: src/components/DialoguesList/DialoguesList.jsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import MessageBubble from '../MessageBubble/MessageBubble';
 
 /**
  * DialoguesList component displays messages in the form of MessageBubbles between users.
- * This component takes in an array of objects (messages) which contain properties sender id, text, 
+ * This component takes in an array of objects (messages) which contain properties sender id, text,
```
Comment (uoa-lzha989) [inline src/components/DialoguesList/DialoguesList.jsx:10]: Shouldn't this comment be in the same line?

**Review by uoa-lzha989 — COMMENTED** [review]
Not a lot to say, just some minor changes that would ideally be made :) otherwise great job!

File: src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap (hunk 2 above)
Comment (colaMeowSY) [reply src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap:29]: This file is generated snapshot after running npm test, I don't think I can change this

File: src/components/DialoguesList/DialoguesList.jsx (hunk 3 above)
Comment (colaMeowSY) [reply src/components/DialoguesList/DialoguesList.jsx:10]: I'll fix the comment as it's not marked as done yet :)

File: src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap (hunk 2 above)
Comment (uoa-lzha989) [reply src/components/DialoguesList/__tests__/__snapshots__/DialoguesList.test.jsx.snap:29]: Lmao sorry brain just blanked on what file I was looking at

**Review by scorpionknifes — APPROVED** [review]

**Review by DanilLinkov — APPROVED** [review]

**Pull request merged**